test/shopify-cli/js_system_test.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+require 'project_types/node/test_helper'
+
+module ShopifyCli
+  class JsSystemTest < MiniTest::Test
+    def setup
+      project_context('app_types', 'node')
+      @system = JsSystem.new(ctx: @context)
+
+      @execution_mock = Minitest::Mock.new
+      @execution_mock.expect(:executed, nil)
+    end
+
+    def test_call_executes_yarn_lambda_if_yarn_is_available
+      @system.stubs(:yarn?).returns(true)
+
+      @system.call(yarn: -> { @execution_mock.executed }, npm: -> { raise Exception })
+
+      @execution_mock.verify
+    end
+
+    def test_call_executes_npm_lambda_if_npm_is_available
+      @system.stubs(:yarn?).returns(false)
+
+      @system.call(yarn: -> { raise Exception }, npm: -> { @execution_mock.executed })
+
+      @execution_mock.verify
+    end
+
+    def test_call_executes_yarn_command_array_if_yarn_is_available
+      yarn_command = %w(install --silent)
+
+      @system.stubs(:yarn?).returns(true)
+      mock_kit_system(JsSystem::YARN_CORE_COMMAND, *yarn_command)
+
+      @system.call(yarn: yarn_command, npm: ['npm'])
+    end
+
+    def test_call_executes_npm_command_array_if_npm_is_available
+      npm_command = %w(install --other)
+
+      @system.stubs(:yarn?).returns(false)
+      mock_kit_system(JsSystem::NPM_CORE_COMMAND, *npm_command)
+
+      @system.call(yarn: ['yarn'], npm: npm_command)
+    end
+
+    def test_call_executes_yarn_command_string_if_yarn_is_available
+      yarn_command = 'install'
+
+      @system.stubs(:yarn?).returns(true)
+      mock_kit_system(JsSystem::YARN_CORE_COMMAND, yarn_command)
+
+      @system.call(yarn: yarn_command, npm: 'npm')
+    end
+
+    def test_call_executes_npm_command_string_if_npm_is_available
+      npm_command = 'install'
+
+      @system.stubs(:yarn?).returns(false)
+      mock_kit_system(JsSystem::NPM_CORE_COMMAND, npm_command)
+
+      @system.call(yarn: 'yarn', npm: npm_command)
+    end
+
+    private
```

Comment:
we don't have tests for any of the class methods

Reply:
added.